whylog/regexassistant/__init__.py

```diff
@@ -0,0 +1,102 @@
+class RegexAssistant(object):
+    """
+    Responsible for creating regex for each line of Rule.
+    RegexAssistant helps user to write regex, it can also proposes regex.
+    One RegexAssistant per one entering Rule.
+    """
+
+    def _obvious_regex(self, text):
+        """
+        Creates regex form text by simple transformation:
+        - add backslash to special characters
+        - text -> '^' + text + '$'
+        :param text:
+        :return:
+        """
+        # not finished yet
+        return '^' + text + '$'
+
+    def __init__(self, line_objects=tuple()):
+        self.regexes = dict(
+            [
+                (line, self._obvious_regex(line.line_content)) for line in line_objects
+            ]
+        )
+        self.history = dict([(line, []) for line in line_objects])
+        pass
+
+    def add_lines(self, line_objects):
+        for line in line_objects:
+            self.regexes[line] = self._obvious_regex(line.line_content)
+            self.history[line] = []
+
+    def remove_lines(self, line_objects):
+        for line in line_objects:
+            del self.regexes[line]
+            del self.history[line]
+
+    def make_groups(self, intervals):
+        """
+        Improves regexes by adding to them regex groups corresponding to params in text.
+        :param intervals: represents interval of text together with line_object
+        """
+        pass
+
+    def remove_groups(self, intervals):
```

Comment:
I don't see that it is impossible, but I will remove it for now - just not to present interface that I am not sure.